examples/gdp/small_lit/nonconvex_HEN.py

```diff
@@ -0,0 +1,94 @@
+""" Example from 'Systematic Modeling of Discrete-Continuous Optimization \
+      Models through Generalized Disjunctive Programming'
+    Ignacio E. Grossmann and Francisco Trespalacios, 2013
+
+    Optimal solution ca. 114,385
+
+    Pyomo model implementation by @RomeoV
+"""
+
+
+from pyomo.environ import (ConcreteModel, Constraint, NonNegativeReals,
+                           Objective, Var, RangeSet, TransformationFactory)
+from pyomo.gdp import Disjunction
+from pyomo.opt import SolverFactory
+
+
+def build_gdp_model():
+
+    # PARAMETERS
+    T1_lo, T1_up = 350., 400.
+    T2_lo, T2_up = 450., 500.
+
+    U = {'1':1.5, '2':0.5, '3':1}
+    FCP = {'hot':10.0, 'cold':7.5}
+    T_in = {'hot':500., 'cold':350., 'cooling':300., 'steam':600.}
+    T_out = {'hot':340., 'cold':560., 'cooling':320., 'steam':600.}
+    Cost = {'cooling':20., 'steam':80.}
+
+    # VARIABLES
+    m = ConcreteModel()
+    m.T1 = Var(domain=NonNegativeReals, bounds=(T1_lo, T1_up))
+    m.T2 = Var(domain=NonNegativeReals, bounds=(T2_lo, T2_up))
+
+    m.exchangers = RangeSet(1,3)
+    m.A  = Var(m.exchangers, domain=NonNegativeReals, bounds=(1e-4,50))
+    m.CP = Var(m.exchangers, domain=NonNegativeReals, bounds=(0,600*pow(50,0.6)+2*46500))
+
+    # OBJECTIVE
+    m.objective = Objective(
+        expr=(sum(m.CP[i] for i in m.exchangers)
+              + FCP['hot']*(m.T1-T_out['hot'])*Cost['cooling']
+              + FCP['cold']*(T_out['cold']-m.T2)*Cost['steam'])
+    )
+
+    # GLOBAL CONSTRAINTS
+    m.constr1 = Constraint(
+        expr=FCP['hot']*(T_in['hot']-m.T1) == m.A[1]*U['1']*((T_in['hot']-m.T2)+(m.T1-T_in['cold']))/2.
+    )
+    m.constr2 = Constraint( # Note the error in the paper in constraint 2
+        expr=FCP['hot']*(m.T1-T_out['hot']) == m.A[2]*U['2']*((T_out['hot']-T_in['cooling'])+(m.T1-T_out['cooling']))/2.
+    )
+    m.constr3 = Constraint(
+        expr=FCP['cold']*(T_out['cold']-m.T2) == m.A[3]*U['3']*((T_out['steam']-m.T2)+(T_in['steam']-T_out['cold']))/2.
+    )
+    m.constr4 = Constraint(
+        expr=FCP['hot']*(T_in['hot']-m.T1) == FCP['cold']*(m.T2-T_in['cold'])
+    )
+
+    # DISJUNCTIONS
+    @m.Disjunction(m.exchangers)
+    def exchanger_disjunction(m, disjctn):
+        return [
+            [m.CP[disjctn] == 2750*pow(m.A[disjctn],0.6)+3000,
+             0. <= m.A[disjctn], m.A[disjctn] <= 10.],
+            [m.CP[disjctn] == 1500*pow(m.A[disjctn],0.6)+15000,
+             10. <= m.A[disjctn], m.A[disjctn] <= 25.],
+            [m.CP[disjctn] == 600*pow(m.A[disjctn],0.6)+46500,
+             25. <= m.A[disjctn], m.A[disjctn] <= 50.]
```

Comment:
When using A_lo = 0, an error can occurr. Originally I thought it might
be due to how gams handles 'pow(0,n)' vs '0**n'. It turns out though
that the problem lies within MC++, when constructing McCormick
relaxations (that are used when solving globally with 'gdpopt'). I have 'fixed it' (aka circumvented it) by setting A_lo to
a small value. Nonetheless, it is a problem that should be looked into
further.  
Anyways - I have changed 'pow' back to '**' because it wasn't a problem in gams after all.